base_layer/wallet/src/transaction_service/service.rs

```diff
@@ -1682,6 +1687,21 @@ where
                             Some(s) => s,
                         }
                     },
+                    Err(TransactionStorageError::ValueNotFound(_)) => {
```

Comment:
Is this correct? Feels like it might send a false positive

Reply:
I was looking at this code, and added this extra. 
My reasoning was, we don't know about this tx for whatever reason. 
For example, our storage crashed or something. Now someone is asking about the tx and we don't know about, so rather send them a cancel than just ignore the request and they don't have to continue looking at the tx.

Reply:
I believe `Err(_) => return Err(TransactionServiceError::TransactionDoesNotExistError)` (or `Err(TransactionStorageError::ValueNotFound(key))`) when receiving a finalized transaction should result in a recovery attempt for those outputs.

Reply:
Any error there could result in lost funds. We need to do some recovery here to get those funds. At this point, the sender cannot cancel the tx, so we should try and do a recovery to recover that tx as we should have created the keys from our master.